File: src/Java/Programmers/Lv2/PR_완전탐색_카펫.java
package Java.Programmers.Lv2;

import java.util.Arrays;

class Carpet {
    public int[] carpet(int brown, int yellow){
        int[] answer = new int[2];
        int sum = brown + yellow;

        for(int i = 3; i<= sum; i++){
            int row = i; // 세로
            int col = sum / i; // 가로

            if(row>col) continue;

            if((row - 2) * (col - 2) == yellow){
                answer[0] = col;
                answer[1] = row;
                break;
            }
        }
        return answer;
    }

    public static void main(String[] args){
        int brown = 10;
        int yellow = 2;

        Carpet sol = new Carpet();
        System.out.println(Arrays.toString(sol.carpet(brown, yellow)));
    }
}

File: src/Kotlin/PR_완전탐색_카펫.kt
package Kotlin

class Carpet{
    fun carpet(brown: Int, yellow: Int): IntArray{
        val answer = IntArray(2, { 0 })
        val sum = brown + yellow

        for(i in 3..sum){
            val row = i
            val col = sum / i

            if(row > col) continue

            if((row - 2) * (col - 2) == yellow){
                answer[0] = col
                answer[1] = row
                break
            }
        }
        return answer
    }

}
fun main(){
    val brown = 10
    val yellow = 2
    val sol = Carpet()

    println(sol.carpet(brown, yellow).contentToString())
}